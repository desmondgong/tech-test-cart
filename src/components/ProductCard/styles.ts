import { Theme } from '@material-ui/core';
import { BaseCSSProperties } from '@material-ui/styles';

export interface StyleProps {
  root: BaseCSSProperties;
  content: BaseCSSProperties;
}
const styles = (theme: Theme): StyleProps => ({
  root: {
    width: 280,
    margin: theme.spacing(2),
    boxShadow: '1px 1px 5px #dde1f8',
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
export default styles;
