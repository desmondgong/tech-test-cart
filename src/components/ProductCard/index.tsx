import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
import { makeStyles } from '@material-ui/core/styles';
import { formatPrice } from '../../utils';
import { Product } from '../../types';
import styles from './styles';

export interface Props extends Product {
  onAdd?: () => void;
}

const useStyles = makeStyles(styles);

const ProductCard = (props: Props): JSX.Element => {
  const { name, price, onAdd } = props;
  const classes = useStyles(props);
  return (
    <Card className={classes.root}>
      <CardContent className={classes.content}>
        <Typography variant="body1">{name}</Typography>
        <Typography variant="body1">{formatPrice(price)}</Typography>
        <Tooltip title="add to cart">
          <IconButton color="primary" onClick={onAdd}>
            <AddShoppingCartIcon />
          </IconButton>
        </Tooltip>
      </CardContent>
    </Card>
  );
};

ProductCard.defaultProps = {
  onAdd: () => {},
};

export default ProductCard;
