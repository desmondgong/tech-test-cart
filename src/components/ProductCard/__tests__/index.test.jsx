import React from 'react';
import renderer from 'react-test-renderer';
import { renderWithTheme } from '../../../tests/testUtils';
import ProductCard from '..';

const props = {
  id: '1',
  name: 'Sledgehammer',
  price: 125.75,
};

it('Matches the snapshots', () => {
  const component = renderer.create(<ProductCard {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly', () => {
  const { getByText } = renderWithTheme(<ProductCard {...props} />);

  expect(getByText(props.name)).toBeInTheDocument();
});
