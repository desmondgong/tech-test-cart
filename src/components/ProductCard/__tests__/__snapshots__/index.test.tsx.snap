// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshots 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root makeStyles-content-2"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Sledgehammer
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      $125.75
    </p>
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="add to cart"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
